shared/crypto/slip.go

```diff
@@ -0,0 +1,156 @@
+package crypto
```

Comment:
Can you add `slip_test.go` (positive & negative test cases) for https://github.com/satoshilabs/slips/blob/master/slip-0010.md to make sure it works.

Sleep need a bit more time to understand the spec and verify.